File: ios/ReservationApp/ReservationApp/Core/Network/NetworkError.swift
//
//  NetworkError.swift
//  ReservationApp
//
//  Created by Mehmet Ali Kısacık on 13.11.2025.
//

import Foundation

enum NetworkError: LocalizedError {
    case invalidURL
    case noData
    case decodingError(Error)
    case serverError(Int, String?)
    case unauthorized
    case networkError(Error)
    case timeout
    case unknown
    
    var errorDescription: String? {
        switch self {
        case .invalidURL:
            return "Geçersiz URL"
        case .noData:
            return "Veri alınamadı"
        case .decodingError(let error):
            return "Veri çözümlenemedi: \(error.localizedDescription)"
        case .serverError(let code, let message):
            return message ?? "Sunucu hatası: \(code)"
        case .unauthorized:
            return "Yetkisiz erişim. Lütfen tekrar giriş yapın."
        case .networkError(let error):
            return "Ağ hatası: \(error.localizedDescription)"
        case .timeout:
            return "İstek zaman aşımına uğradı. Lütfen tekrar deneyin."
        case .unknown:
            return "Bilinmeyen bir hata oluştu"
        }
    }
}

File: android/core/network/src/main/java/com/reservationapp/core/network/NetworkError.kt
package com.reservationapp.core.network

sealed class NetworkError(override val message: String = "") : Exception(message) {
    data class HttpError(val code: Int, override val message: String) : NetworkError(message)
    data class NetworkException(val throwable: Throwable) : NetworkError(throwable.message ?: "Network error")
    object UnknownError : NetworkError("Unknown error")
    object Unauthorized : NetworkError("Unauthorized")
    object NotFound : NetworkError("Not found")
    object ServerError : NetworkError("Server error")
}

